app/nav_bar/page.jsx

```diff
@@ -0,0 +1,93 @@
+"use client";
+import { useState } from "react";
+import {
+  AppBar,
+  Toolbar,
+  Typography,
+  Button,
+  IconButton,
+  Drawer,
+  List,
+  ListItem,
+  ListItemText,
+  Box,
+} from "@mui/material";
+import MenuIcon from "@mui/icons-material/Menu";
+import { useTheme, useMediaQuery } from "@mui/material";
+import Link from "next/link";
+
+export default function Navbar() {
+  const [drawerOpen, setDrawerOpen] = useState(false);
+  const theme = useTheme();
+  const isMobile = useMediaQuery(theme.breakpoints.down("md"));
+
+  const toggleDrawer = () => {
+    setDrawerOpen(!drawerOpen);
+  };
+  const navItems = [
+    { text: "Home", link: "/nav_bar" },
+    { text: "About", link: "/nav_bar/about" },
+    { text: "Contact", link: "/nav_bar/contact" },
+  ];
+  return (
+    <AppBar
+      position="static"
+      sx={{
+        background: "linear-gradient(135deg, #6a11cb 0%, #2575fc 100%)",
+        boxShadow: "none",
+        padding: "0.5rem 0",
+      }}
+    >
+      <Toolbar>
+        <Typography
+          variant="h6"
+          sx={{ flexGrow: 1, fontWeight: "bold", letterSpacing: 1.5 }}
+        >
+          StufHub
+        </Typography>
+
+        {isMobile ? (
+          <>
+            <IconButton
+              edge="start"
+              color="inherit"
+              aria-label="menu"
+              sx={{ mr: 2 }}
+              onClick={toggleDrawer}
+            >
+              <MenuIcon sx={{ fontSize: 30, color: "#fff" }} />
+            </IconButton>
+            <Drawer anchor="right" open={drawerOpen} onClose={toggleDrawer}>
+              <Box
+                sx={{ width: 250, padding: "1rem" }}
+                role="presentation"
+                onClick={toggleDrawer}
+                onKeyDown={toggleDrawer}
+              >
+                <List>
+                  {navItems.map((item) => (
+                    <ListItem button key={item.text}>
+                      <Link href={item.link} passHref>
+                        <ListItemText primary={item.text} />
+                      </Link>
+                    </ListItem>
+                  ))}
+                </List>
+              </Box>
+            </Drawer>
+          </>
+        ) : (
```

Comment:
Very well done! I only think it would be a lot more readable if this was split into more subcomponents with props passed into it.